packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2835,7 +2988,10 @@ const arraySubsequentOpenBracket = stringToPrecomputedChunk(',[');
 const arrayInterstitial = stringToPrecomputedChunk(',');
 const arrayCloseBracket = stringToPrecomputedChunk(']');
 
-function writeStyleResourceDependencies(
+// This function writes a 2D array of strings to be embedded in javascript.
+// E.g.
+//  [["JS_escaped_string1", "JS_escaped_string2"]]
+function writeStyleResourceDependenciesInJS(
```

Comment:
This felt pretty messy to write. Ideally, `writeStyleResourceDependenciesInJS` and `writeStyleResourceDependenciesInAttr` would each live in a module and use local helpers.

Do we prefer keeping helper functions all in the same file? If not, where would be the best directory for these helper files?

Reply:
embrace the thousand line files! :) Yeah I had the same impulse when I first started but there is something nice about all the code you generally need to consider being in one place. In fact one could argue float itself should just be copied into this file. There are actually some places the API is more awkward because it isn't defined in the same scope (I forget the details atm but it did come up during review)